app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,38 @@
+class TagsController < ApplicationController
+  before_action :set_tag, only: [ :edit, :update, :destroy ]
+
+  def index
+    @tags = Current.user.tags.left_outer_joins(:taggings)
+      .group('tags.id')
+      .select('tags.*, COUNT(taggings.id) as taggings_count')
+      .order('taggings_count DESC, tags.name ASC')
+  end
+
+  def edit
+  end
+
+  def update
+    if @tag.update(tag_params)
+      redirect_to tags_path, notice: 'Tag was successfully updated.'
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @tag.destroy
+    redirect_to tags_path, notice: 'Tag was successfully deleted.'
```

Comment:
```suggestion
    if @tag.destroy
      redirect_to tags_path, success: 'Tag was successfully deleted.'
    else
      redirect_to tags_path, error: 'Failed to delete tag.'
    end
```

Add a fallback if the destroy fails for some reason, and to be consistent with other places in the codebase. Also, flash type consistency again